client/src/main/java/io/pravega/client/stream/impl/Pinger.java

```diff
@@ -79,13 +74,11 @@ void stopPing(UUID txnID) {
         }
     }
 
-    private long getPingInterval(long txnLeaseMillis) {
-        double pingInterval = txnLeaseMillis * PING_INTERVAL_FACTOR;
-        if (pingInterval < MINIMUM_PING_INTERVAL_MS) {
-            log.warn("Transaction ping interval is less than 10 seconds(lower bound)");
-        }
-        //Ping interval cannot be less than KeepAlive task interval of 10seconds.
-        return Math.max(MINIMUM_PING_INTERVAL_MS, (long) pingInterval);
+    private long getPingInterval() {
+        //Provides a good number of attempts: 1 for <4s, 2 for <9s, 3 for <16s, 4 for <25s, ... 10 for <100s
+        //while at the same time allowing the interval to grow as the timeout gets larger.
+        double targetNumPings = Math.sqrt(txnLeaseMillis / 1000.0);
+        return Math.round(txnLeaseMillis / targetNumPings);
```

Comment:
This could end up being 0, correct? (If txnLeaseMillis<500).

Should we ensure that it is at least 1?

I know you ensure that it's at least 10000, but that's in a different file, so it wouldn't hurt to be on the safe side here.